### PR TITLE
Fix PaginationToken Circe codecs

### PR DESCRIPTION
**Pull request by pomadchin**
## Overview

This PR fixes PaginationToken circe codecs. Without it post requests with the passed `next` token doesn't work. I discovered this issue in terms of https://github.com/azavea/stac4s/pull/327

### Checklist

- [x] New tests have been added or existing tests have been modified

### CODE REVIEW

**Comment by jisantuc** [comment]
Could this get a test? I think it wouldn't be too tough to add similar to here --

https://github.com/azavea/franklin/blob/master/application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala#L73-L82

The testing procedure would be like:

- generate two items
- send the POST search request with a `limit` of 1
- ensure that you can use the `next` link to send another valid POST request

**Comment by pomadchin** [comment]
@jisantuc ok, will push it, but I want to admit, that because codecs are incorrect test would work anyway. That's due to the fact that both default encoder and decoder are incorrect (they encode PaginationToken into a JSON object not into a string).

File: application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala [hunk 1]
```diff
@@ -117,6 +120,51 @@ class SearchServiceSpec
...
+        resource.use {
+          case (collection, _) =>
+            val inclusiveParams =
+              FiltersFor.inclusiveFilters(collection).copy(limit = NonNegInt.unsafeFrom(1).some)
+            val result1 = getSearchCollection(inclusiveParams)
+
+            val result2 = result1
+              .flatMap {
+                _.traverse { r =>
+                  /** This line intentionally decodes next token [[String]] into [[PaginationToken]] */
```
Comment (pomadchin) [inline application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala:148]: That's a pretty important comment

File: application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala [hunk 2]
```diff
@@ -117,6 +120,51 @@ class SearchServiceSpec
...
+          (for {
+            resp    <- testServices.searchService.routes.run(request)
+            decoded <- OptionT.liftF { resp.as[StacSearchCollection] }
+          } yield decoded).value
+        }
+
+        resource.use {
+          case (collection, _) =>
+            val inclusiveParams =
+              FiltersFor.inclusiveFilters(collection).copy(limit = NonNegInt.unsafeFrom(1).some)
```
Comment (jisantuc) [inline application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala:142]: You can import `eu.timepit.refined.auto._` and safely use a literal `1` here if you want. Though I'm not sure how that will play with `.some` tbh

File: application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala [hunk 3]
```diff
@@ -117,6 +120,51 @@ class SearchServiceSpec
...
+            val result1 = getSearchCollection(inclusiveParams)
+
+            val result2 = result1
+              .flatMap {
+                _.traverse { r =>
```
Comment (jisantuc) [inline application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala:147]: with the `.map(_.flatten)` at the end, can this be a `flatTraverse`?

**Review by jisantuc — COMMENTED** [review]
> both default encoder and decoder are incorrect (they encode PaginationToken into a JSON object not into a string).

I don't think that's quite right unless you're talking about the circe codecs. The codec for the query param decodes from a b64 encoded string --

https://github.com/azavea/franklin/blob/c67f29092dfc7aea3cbe457d76d70f59b1a6a43c/application/src/main/scala/com/azavea/franklin/api/schemas/package.scala#L85-L86

I think actually the derived decoders just shouldn't be there, since we never want to output / consume the JSON representation with fields and want it to stay opaque for clients. Maybe I never really understood what you were saying the problem was, but if the circe codec used the same encPaginationToken and decPaginationToken methods, I think everything would be consistent?

**Comment by pomadchin** [comment]
@jisantuc just to be clear, what is the correct `next` Search field representation?

**Comment by pomadchin** [comment]
@jisantuc not PaginationToken was the issue, but more an incorrect decoding of the `SearchFilters`.

File: application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala (hunk 2 above)
Comment (pomadchin) [reply application/src/test/scala/com/azavea/franklin/api/services/SearchServiceSpec.scala]: replaced with `NonNegInt.from(1).toOption` shrug

**Review by jisantuc — APPROVED** [review]
🚀 it took me a minute to understand the problem, but I've confirmed now that you get the same results with:

- posting search filters with the b64 encoded pagination token in the `next` field
- posting empty search filters to the url with the `next` qp set

I also confirmed that the POST with the b64 encoded token to the AVIRIS Franklin deployment is a 400

**Pull request merged**